spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAlterTablePartitionFields.java

```diff
@@ -421,6 +421,31 @@ public void testSparkTableAddDropPartitions() throws Exception {
         "spark table partition should be empty", 0, sparkTable().partitioning().length);
   }
 
+  @Test
+  public void testDropColumnOfOldPartitionFieldV1() {
+    // default table created in v1 format
+    sql(
+        "CREATE TABLE %s (id bigint NOT NULL, ts timestamp, day_of_ts date) USING iceberg PARTITIONED BY (day_of_ts)",
+        tableName);
+
+    sql("ALTER TABLE %s REPLACE PARTITION FIELD day_of_ts WITH days(ts)", tableName);
+
+    sql("ALTER TABLE %s DROP COLUMN day_of_ts", tableName);
```

Comment:
Can we also include a sql to read back the data from the table after the drop? We have noticed the same issue in Trino, but for us the drop column still succeeds, it's the subsequent read operations that start failing

Reply:
Thanks @marton-bod, the `ALTER TABLE` threw an exception before, but you are right that subsequent reads are also failing. Digging into this

Reply:
Hmm, Interesting. This is what I assumed would happen and went with spec change (disruptive change)

@marton-bod: I have tagged you in one of the slack discussions. Where I proposed spec change to handle this.  
If Fokko doesn't find an easy way to fix this, we can discuss my approach after his experiment.